test/controllers/api/v2/host_collections_controller_test.rb

```diff
@@ -66,6 +67,15 @@ def test_create
       assert_template 'api/v2/host_collections/create'
     end
 
+    def test_validate_max_content_host
+      post :create, :organization_id => @organization,
+                    :host_collection => {:name => 'Collection A', :description => 'Collection A, World Cup 2014',
+                                         :max_content_hosts => 1, :unlimited_content_hosts => false,
+                                         :system_ids => [@system.id, @system_two.id]}
+
+      assert_response 422
```

Comment:
I would probably also check the response or `assigns(:host_collection).errors` to make sure we're setting the right error.